lib/_http_outgoing.js

```diff
@@ -74,13 +72,63 @@ function OutgoingMessage() {
   this.socket = null;
   this.connection = null;
   this._header = null;
-  this._headers = null;
+  this[outHeadersKey] = null;
 
   this._onPendingData = null;
 }
 util.inherits(OutgoingMessage, Stream);
 
 
+Object.defineProperty(OutgoingMessage.prototype, '_headers', {
+  get: internalUtil.deprecate(function() {
+    return this.getHeaders();
+  }, 'msg._headers is deprecated. Use public API methods instead.'),
+  set: internalUtil.deprecate(function(val) {
+    if (val == null) {
+      this[outHeadersKey] = null;
+    } else if (typeof val === 'object') {
```

Comment:
hmm... if the `val` happens to be an Array this would have some odd effects. I don't want this getting any more complicated so perhaps that's ok, but I'd rather not see header name keys like `0: whatever`, `1: value`, etc